### DISCUSSION TITLE
Dynamic Validation Rule Values

### DISCUSSION THREAD

**Discussion by hirenkeraliya** [spatie/laravel-data#51] [Q&A]
What do we do if we need to dynamic values of the `IN` validation rule?

```
#[In(implode(',', array_keys(ImportData::TYPES)))]
public int $typeId,
```

**Comment by rubenvanassche** [comment]
You should use the `rules` method on a data object for more dynamic stuff (https://spatie.be/docs/laravel-data/v1/as-a-data-transfer-object/request-to-data-object#using-validation)
